### PR TITLE
[15.0][MIG] mis_builder_demo,mis_builder_budget: Migration to 15.0

### PR DESCRIPTION
**Pull request by yankinmax**

### CODE REVIEW

File: mis_builder_budget/__manifest__.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "name": "MIS Builder Budget",
     "summary": """
         Create budgets for MIS reports""",
-    "version": "14.0.4.0.0",
+    "version": "15.0.1.0.0",
```
Comment (sbidoul) [inline mis_builder_budget/__manifest__.py]:
```suggestion
    "version": "15.0.4.0.0",
```

File: mis_builder_budget/models/mis_budget_abstract.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def copy(self, default=None):
             default = {}
         if "name" not in default:
             default["name"] = _("%s (copy)") % self.name
-        return super(MisBudgetAbstract, self).copy(default=default)
+        return super().copy(default=default)
```
Comment (sbidoul) [inline mis_builder_budget/models/mis_budget_abstract.py]: Can you do these changes that are not related to migration in a separate commit (or better, a separate PR), so they can be backported easily ?

File: mis_builder_budget/models/mis_budget_by_account_item.py [hunk 3]
```diff
@@ -70,10 +70,11 @@ def _prepare_overlap_domain(self):
             # overlapping budget items is only possible for budget by account items
             # and kpi budget items.
             return [("id", "=", 0)]
-        domain = super(MisBudgetByAccountItem, self)._prepare_overlap_domain()
+        domain = super()._prepare_overlap_domain()
         domain.extend([("account_id", "=", self.account_id.id)])
         return domain
 
+    # pylint: disable=W8110
```
Comment (sbidoul) [inline mis_builder_budget/models/mis_budget_by_account_item.py]: What is W8110 ? Can you use the human-readable code ?

File: mis_builder_demo/__manifest__.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
     "name": "MIS Builder Demo",
     "summary": """
         Demo addon for MIS Builder""",
-    "version": "14.0.3.1.3",
+    "version": "15.0.1.0.0",
```
Comment (sbidoul) [inline mis_builder_demo/__manifest__.py]:
```suggestion
    "version": "15.0.3.1.0",
```

**Comment by yankinmax** [comment]
@sbidoul updated with your suggestions

**Comment by yankinmax** [comment]
@misern2 @antoniocanovas can you pls take a look?

**Comment by yankinmax** [comment]
@sbidoul I rebased this PR.

File: mis_builder_budget/__manifest__.py [hunk 5]
```diff
@@ -5,7 +5,7 @@
     "name": "MIS Builder Budget",
     "summary": """
         Create budgets for MIS reports""",
-    "version": "14.0.4.0.0",
+    "version": "15.0.4.0.0",
```
Comment (leemannd) [inline mis_builder_budget/__manifest__.py:8]:
```suggestion
    "version": "15.0.1.0.0",
```

File: mis_builder_demo/__manifest__.py [hunk 6]
```diff
@@ -5,7 +5,7 @@
     "name": "MIS Builder Demo",
     "summary": """
         Demo addon for MIS Builder""",
-    "version": "14.0.3.1.3",
+    "version": "15.0.3.1.0",
```
Comment (leemannd) [inline mis_builder_demo/__manifest__.py:8]:
```suggestion
    "version": "15.0.1.0.0",
```

File: mis_builder_budget/__manifest__.py (hunk 5 above)
Comment (yankinmax) [reply mis_builder_budget/__manifest__.py:8]: this should be like that. @sbidoul pointed here: https://github.com/OCA/mis-builder/pull/406#discussion_r794293959

File: mis_builder_demo/__manifest__.py (hunk 6 above)
Comment (yankinmax) [reply mis_builder_demo/__manifest__.py:8]: @leemannd same here https://github.com/OCA/mis-builder/pull/406#discussion_r794295627

File: mis_builder_budget/__manifest__.py (hunk 5 above)
Comment (leemannd) [reply mis_builder_budget/__manifest__.py:8]: Ok, strange. I don't understand why

**Review by leemannd — APPROVED** [review]
Seems legit, but I'm unsure about the module version number

File: mis_builder_budget/__manifest__.py (hunk 5 above)
Comment (sbidoul) [reply mis_builder_budget/__manifest__.py:8]: @leemannd because I try to maintain feature parity across Odoo versions, and the major and minor version number communicates that. TBH, resetting to 1.0 when we change Odoo version is what looks *really* strange to me, but :shrug: there are fights I have abandonned.

Comment (leemannd) [reply mis_builder_budget/__manifest__.py:8]: @sbidoul Thanks for the answer. I was surprised has it is not commonly done in other modules in OCA.
On this topic I understand both point of views. But IMO your method is best suited if there is a maintainer that has the time to care (and has the time) about _maintaining_ the code as a library. Which seems to be the case

Comment (sbidoul) [reply mis_builder_budget/__manifest__.py:8]: @leemannd yes, exactly that. And I sure wish there were more real maintainers in OCA.

**Comment by yankinmax** [comment]
@sbidoul is it ok now for you? can you submit your approval? or you want me to ask for more reviews?

**Comment by sbidoul** [comment]
Yes, looks good, thanks!

/ocabot merge patch

File: mis_builder_demo/__manifest__.py (hunk 6 above)
Comment (leemannd) [reply mis_builder_demo/__manifest__.py:8]: @yankinmax You can ignore my comment about the version numbers
